src/store/hotspots/hotspotsSlice.ts

```diff
@@ -1,7 +1,40 @@
 import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
 import { Hotspot, Sum } from '@helium/http'
 import Balance, { CurrencyType, NetworkTokens } from '@helium/currency'
+import { orderBy, sortBy } from 'lodash'
 import { getHotspotRewardsSum, getHotspots } from '../../utils/appDataClient'
+import { distance, LocationCoords } from '../../utils/location'
+
+export enum HotspotSort {
+  New = 'new',
+  Near = 'near',
+  Earn = 'earn',
+  Offline = 'offline',
+}
+
+type Rewards = Record<string, Sum>
+
+type SorterContext = { rewards?: Rewards; location?: LocationCoords }
+type HotspotSorter = (hotspots: Hotspot[], context?: SorterContext) => Hotspot[]
+const hotspotSorters: Record<HotspotSort, HotspotSorter> = {
```

Comment:
Nice, I like how you set this up

Reply:
thanks, it was a little awkward getting some of the additional context in there, like the rewards and location, but seems to work ok